File: ios-native/IronCore/Models/Workout.swift
import Foundation
import FirebaseFirestore

/// Workout log at `users/{userId}/workouts/{workoutId}`
struct Workout: Codable, Identifiable {
    @DocumentID var id: String?
    var date: String            // "YYYY-MM-DD"
    var userId: String
    var exercises: [Exercise]
    var duration: Int?          // seconds
    @ServerTimestamp var createdAt: Timestamp?
}

struct Exercise: Codable {
    var name: String
    var sets: [ExerciseSet]

    /// Total volume = sum(weight * reps) across all sets
    var totalVolume: Double {
        sets.reduce(0) { $0 + ($1.w * Double($1.r)) }
    }
}

struct ExerciseSet: Codable {
    var w: Double   // weight
    var r: Int      // reps
}

File: android-native/app/src/main/java/com/ironcore/fit/data/model/Workout.kt
package com.ironcore.fit.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

/**
 * Firestore path: users/{uid}/workouts/{workoutId}
 *
 * Matches the React app's Firestore schema exactly so users can
 * switch between web and native seamlessly.
 */
data class Workout(
    val id: String = "",
    val name: String = "",
    val type: String = "strength",        // "strength" | "cardio" | "flexibility"
    val exercises: List<WorkoutExercise> = emptyList(),
    val duration: Long = 0,               // seconds
    val caloriesBurned: Int = 0,
    val xpEarned: Int = 0,
    val date: String = "",                // "YYYY-MM-DD"
    val startedAt: Timestamp? = null,
    @ServerTimestamp val completedAt: Timestamp? = null
)

data class WorkoutExercise(
    val name: String = "",
    val muscleGroup: String = "",
    val isCustom: Boolean = false,
    val sets: List<ExerciseSet> = emptyList()
)

data class ExerciseSet(
    val weight: Double = 0.0,
    val reps: Int = 0,
    val rpe: Int? = null,
    val completed: Boolean = false
)

/**
 * Firestore path: workoutTemplates/{templateId}
 *
 * User-created workout templates that can be reused to start new workouts.
 */
data class WorkoutTemplate(
    val id: String = "",
    val userId: String = "",
    val name: String = "",
    val type: String = "strength",
    val exercises: List<WorkoutExercise> = emptyList(),
    @com.google.firebase.firestore.ServerTimestamp val createdAt: com.google.firebase.Timestamp? = null
)
